test/integration/oauth_server_headers_test.go

```diff
@@ -95,19 +113,30 @@ func checkNewReqHeaders(t *testing.T, rt http.RoundTripper, checkUrl string) {
 
 		// non-security headers, should not change
 		// adding items here should be validated to make sure they do not conflict with any security headers
-		"Content-Type": "text/html; charset=utf-8",
+		// <no items currently>
 	} {
 		// use set so we get the canonical form of these headers
 		allHeaders.Set(key, val)
 	}
 
 	// these headers can change per request and are not important to us
 	// only add items to this list if they cannot be statically checked above
-	ignoredHeaders := []string{"Audit-Id", "Date", "Content-Length", "Location"}
+	ignoredHeaders := []string{"Audit-Id", "Date", "Content-Type", "Content-Length", "Location"}
 	for _, h := range ignoredHeaders {
 		resp.Header.Del(h)
 	}
 
+	// tolerate additional header set by osin library code
+	expires := resp.Header["Expires"]
+	if len(expires) == 2 && expires[1] == "Fri, 01 Jan 1990 00:00:00 GMT" {
```

Comment:
wat? I know osin library is crazy but what's going on here?

Reply:
It insists on blindly adding those headers -.-